### PR TITLE
2106 refactor island mode

### LINKED ISSUE #2106: Refactor `/api/island-mode`
# Refactor

## Component(s) to be refactored

- `/api/island-mode`

## Tasks
- [x] Rename to `/api/island/mode` (0d) @ilija-lazoroski 
- [x] Change `POST` to `PUT` (0d) @ilija-lazoroski 
- [x] Use `HTTPStatus` enum for return codes (0d) @ilija-lazoroski 
  - [x] See if you can just return the status with no body
  - [x] Use 204 instead of 200 for `PUT`
- [ ] Get rid of `['mode']` in `api/island/mode` and all uses. (0.25d) @ilija-lazoroski 
    - [ ] Add `IslandMode` to `resources/__init__.py` and shorten all imports

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes #2106 .

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

File: monkey/monkey_island/cc/resources/island_mode.py [hunk 1]
```diff
@@ -12,30 +13,26 @@
...
+            return {}, HTTPStatus.BAD_REQUEST
         except ValueError:
-            return make_response({}, 422)
+            return {}, HTTPStatus.UNPROCESSABLE_ENTITY
 
     @jwt_required
     def get(self):
         island_mode = self._island_mode_service.get_mode()
-        return make_response({"mode": island_mode.value}, 200)
+        return make_response({"mode": island_mode.value}, HTTPStatus.OK)
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/resources/island_mode.py:38]: While we're working on this (maybe in a follow-up PR), can we just return a string? The dict is redundant; the resource is already called "mode".

```suggestion
        return make_response(island_mode.value, HTTPStatus.OK)
```

File: monkey/monkey_island/cc/ui/src/components/IslandHttpClient.tsx [hunk 2]
```diff
@@ -12,11 +12,11 @@ export class Response{
 }
 
 class IslandHttpClient extends AuthComponent {
-  post(endpoint: string, contents: any): Promise<Response>{
+  put(endpoint: string, contents: any): Promise<Response>{
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/IslandHttpClient.tsx:15]: Nothing else uses this?

File: monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx [hunk 3]
```diff
@@ -96,27 +96,27 @@ const IslandResetModal = (props: Props) => {
...
           }})
       .then(res => {
-        if (res.status === 200) {
-            return auth.authFetch('/api/island-mode', {method: 'POST', body: '{"mode": "unset"}'})
+        if (res.ok) {
+            return auth.authFetch('/api/island/mode', {method: 'PUT', body: '{"mode": "unset"}'})
         }})
       .then(res => {
-        if (res.status !== 200) {
+        if (!res.ok) {
```
Comment (mssalvatore) [inline monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx:119]: `PUT /api/island/mode` returns 204, not 200. Does `res.ok` capture this?

File: monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py [hunk 4]
```diff
@@ -36,22 +36,22 @@ def flask_client(build_flask_client):
     [IslandMode.RANSOMWARE.value, IslandMode.ADVANCED.value, IslandMode.UNSET.value],
 )
 def test_island_mode_post(flask_client, mode):
-    resp = flask_client.post(
+    resp = flask_client.put(
         IslandModeResource.urls[0], data=json.dumps({"mode": mode}), follow_redirects=True
     )
-    assert resp.status_code == 200
+    assert resp.status_code == 204
```
Comment (mssalvatore) [inline monkey/tests/unit_tests/monkey_island/cc/resources/test_island_mode.py:42]: Use `HTTPStatus`.

**Review by mssalvatore — CHANGES_REQUESTED** [review]

File: monkey/monkey_island/cc/ui/src/components/IslandHttpClient.tsx (hunk 2 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/IslandHttpClient.tsx:15]: Nothing that I could find.

File: monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx (hunk 3 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx:119]: `res.ok` is if the response was successful, I am not sure if we should rely on the status code as previously was comparing `200` but we return `204` and while the response was `ok` the reseting of the island mode was not working.

File: monkey/monkey_island/cc/resources/island_mode.py (hunk 1 above)
Comment (ilija-lazoroski) [reply monkey/monkey_island/cc/resources/island_mode.py:38]: I think with only a string we can get rid of the `make_response`. I will add another task in the issue.

File: monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx (hunk 3 above)
Comment (mssalvatore) [reply monkey/monkey_island/cc/ui/src/components/ui-components/IslandResetModal.tsx:119]: We're good: 
image

**Pull request closed without merging**